File: bench/sieve.lua
function sieve(n, primes)
    local sqrt_n = math.sqrt(n)
    for p=2,sqrt_n do
        if(primes[p] == true) then
            for i = p*p,n,p do
                primes[i] = false
            end
        end
    end

    local count = 0
    for i = 2,n do
        if (primes[i] == true) then
            count = count + 1
        end
    end
    return count
end

local primes = {}
for i =1,5000000 do
    primes[i] = true
end

local time = os.clock()
print(sieve(5000000, primes))
print((os.clock()-time)*1000)

File: bench/sieve.js
function sieve(primes) {
    let n = primes.length - 1;
    const sqrt_n = Math.sqrt(n)
    for (let p = 2; p <= sqrt_n; p++) {
        if (primes[p] === true) {
            for (let i = p * p; i <= n; i += p)
                primes[i] = false
        }
    }

    let count = 0
    for (let i = 2; i <= n; i++) {
        if (primes[i] === true)
            count++
    }
    return count
}
let arr = new Array(5_000_001).fill(true);
let time = performance.now()
console.log(sieve(arr))
console.log(performance.now() - time)
